dm/checker/checker.go

```diff
@@ -246,13 +248,37 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 		}
 	}
 
-	if checkingShard {
-		for name, shardingSet := range sharding {
-			if shardingCounter[name] <= 1 {
-				continue
+	instance := c.instances[0]
+	if checkingShard && instance.cfg.Mode != config.ModeIncrement {
```

Comment:
I forget if we should check shard table structure for pessimistic incremental task 😭 currently your code will skip adding checker for pessimistic incremental task . Could you left some comment of decision?

Reply:
https://github.com/pingcap/tiflow/blob/e2d529cc9887ab61063555dbbbf1d9f9765581e3/dm/docs/RFCS/20211130_enhanced_pre_checker.md?plain=1#L53